lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -248,7 +248,28 @@ def get_or_create_cached_value():
                         return_value = func(*args, **kwargs)
 
                 messages = cached_func.message_call_stack._most_recent_messages
-                cache.write_result(value_key, return_value, messages)
+                try:
+                    cache.write_result(value_key, return_value, messages)
+                except TypeError as err:
+                    if (
+                        return_value.__module__ == "snowflake.snowpark.session"
+                        and type(return_value).__name__ == "Session"
+                    ):
+
+                        class SnowparkSessionInMemoDecoratorError(TypeError):
```

Comment:
I think it would be great to move `SnowparkSessionInMemoDecoratorError` class definition to https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/errors.py